js/main.ts

```diff
@@ -79,10 +79,23 @@ export default function denoMain() {
   log("argv", argv);
   Object.freeze(argv);
 
-  const inputFn = argv[0];
-  if (!inputFn) {
-    console.log("No input script specified.");
-    os.exit(1);
+  let inputFn = argv[0];
+
+  if (startResMsg.depfileFlag() && argv.length > 1) {
+    inputFn = argv[1];
+    // TODO: should probably be a function
+    const deps = compiler.placeholderTODO(inputFn, `${cwd}/`);
+    const output: string[] = [];
+    for (const moduleFileName in deps) {
+      const moduleDependencies = deps[moduleFileName] || [];
+      output.push(`${moduleFileName}: ${moduleDependencies.join(" ")}`);
+    }
+    const enc = new TextEncoder();
+    const data = enc.encode(output.join("\n"));
+    const filename = argv[0];
```

Comment:
This isn't a very robust way of getting the flag argument. Now that I think of it, we don't have any flags which take arguments yet, so it may require some preliminary refactoring of the flag parsing code to allow that (eg use https://github.com/rust-lang-nursery/getopts).

Reply:
Should I wait for someone a bit more experienced to take this on or should I give it a shot? I don't mind either way.